gulpfile.js

```diff
@@ -0,0 +1,181 @@
+const gulp = require("gulp");
+const babel = require("gulp-babel");
+const uglify = require("gulp-uglify");
+const rename = require("gulp-rename");
+const sourcemaps = require("gulp-sourcemaps");
+const replace = require("gulp-replace");
+const debug = require("gulp-debug");
+const util = require("gulp-util");
+
+const { rollup } = require("rollup");
+const rollupBabel = require("rollup-plugin-babel");
+const rollupNodeResolve = require("rollup-plugin-node-resolve");
+const rollupCommonjs = require("rollup-plugin-commonjs");
+const rollupJson = require("rollup-plugin-json");
+const rollupReplace = require("rollup-plugin-replace");
+const rollupUglify = require("rollup-plugin-uglify");
+
+const FORMAT_ESM = "esm";
+const FORMAT_CJS = "cjs";
+const FORMAT_UMD = "umd";
+
+const MODULE_NAME = "Eventbrite";
+
+const SOURCE_ENTRY = "src/index.ts";
+
+const FILES_TO_BUILD = [
+  // include all the JavaScript files in src/ directory
+  "src/**/*.@(ts|js)",
+
+  // but exclude the test files
+  "!src/**/__tests__/**"
+];
+
+// When transpiling to ES format, we still use the `env` preset
+// and we want everything transpiled EXCEPT modules
+const ESM_ENV_PRESET = ["@babel/env", { modules: false }];
+
+// When transpiling to UMD, we need the UMD transform plugin.
+// Need to explicitly list the globals unfortunately
+const UMD_TRANSFORM_PLUGIN = [
+  "@babel/plugin-transform-modules-umd",
+  {
+    globals: {
+      index: MODULE_NAME,
+      "isomorphic-fetch": "fetch"
+    },
+    exactGlobals: true
+  }
+];
+
+const _getBabelConfig = format => ({
+  babelrc: false,
+
+  presets: [
+    format === FORMAT_ESM ? ESM_ENV_PRESET : "@babel/env",
+    "@babel/typescript"
+  ],
+  plugins: [
+    "@babel/proposal-class-properties",
+    "@babel/proposal-object-rest-spread",
+    "@babel/plugin-external-helpers",
+    ...(format === FORMAT_UMD ? [UMD_TRANSFORM_PLUGIN] : [])
+  ]
+});
+
+const _getBabelStream = format =>
+  gulp
+    // get a stream of the files to transpile
+    .src(FILES_TO_BUILD)
+    // initialize the sourcemaps (used by UMD only)
+    .pipe(format === FORMAT_UMD ? sourcemaps.init() : util.noop())
+    // do the appropriate babel transpile (this is a copy from package.json)
+    .pipe(babel(_getBabelConfig(format)));
+
+const _genUmd = (minify = false) =>
+  _getBabelStream(FORMAT_UMD)
+    // If you're using UMD, you probably don't have `process.env.NODE_ENV` so, we'll replace it.
+    // If you're using the unminified UMD, you're probably in DEV
+    // If you're using the unminified UMD, you're probably in production
+    .pipe(
+      replace(
+        "process.env.NODE_ENV",
+        JSON.stringify(minify ? "production" : "development")
```

Comment:
actually instead you could set the parameter to `environment = 'development'` and then do `process.env.NODE_ENV || environment` right?

Reply:
Whoops! I totally forgot to add the conditional minification!

```
    .pipe(uglify())
    .pipe(rename({ extname: ".min.js" }))
```

I know that's not what you meant, but thanks for catching! :)